pkg/libvirt/libvirt.go

```diff
@@ -651,12 +574,21 @@ func (d *Driver) GetIP() (string, error) {
 	if s != state.Running {
 		return "", errors.New("host is not running")
 	}
-	mac, err := d.getMAC()
+	ifaces, err := d.vm.ListAllInterfaceAddresses(libvirt.DOMAIN_INTERFACE_ADDRESSES_SRC_LEASE)
 	if err != nil {
 		return "", err
 	}
-
-	return d.getIPByMacFromSettings(mac)
+	if len(ifaces) != 1 {
+		log.Errorf("unexpected number of interfaces for domain %s", d.MachineName)
+		return "", nil
+	}
+	if len(ifaces[0].Addrs) != 1 {
+		log.Errorf("unexpected number of IPs for domain %s", d.MachineName)
```

Comment:
I'd also do some filtering here as maybe on some setups we'll be getting both ipv4 and ipv6 addresses? Don't error out, just return the first ipv4 address, with a warning that there were multiple addresses set on the interface?